### PR TITLE
[Mod] Send ban reason on Temp Bans

### LINKED ISSUE #4836: [Mod] Send ban reason on Temp Bans
**What component of Red (cog, command, API) would you like to see improvements on?**

Mod

**Describe the enhancement you're suggesting.**

Send the ban reason when a user is temporarily banned

### PR DESCRIPTION
**Pull request by Kreusada**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes

Closes #4836. Tempbans will DM the user with the reason specified if any

### CODE REVIEW

**Review by Kowlin — COMMENTED** [review]
From a quick glance at it this does not handle whenever the bot *should* send it's reason via the accompany guild setting

**Review by Naseer502 — DISMISSED** [review]

**Review by Drapersniper — CHANGES_REQUESTED** [review]
If adding more config calls in a command, you should simplify these by only calling config once in a command/function/code path.

File: redbot/cogs/mod/kickban.py [hunk 1]
```diff
@@ -638,16 +640,21 @@ async def tempban(
...
-                )
-            await member.send(msg)
+        if settings["dm_on_kickban"]:
+            with contextlib.suppress(discord.HTTPException):
+                # We don't want blocked DMs preventing us from banning
+                msg = _(
+                    "You have been temporarily banned from {server_name} until {date}."
+                ).format(server_name=guild.name, date=unban_time.strftime("%m-%d-%Y %H:%M:%S"))
+                if reason:
+                    msg += _("\n\n**Reason:** {reason})").format(
```
Comment (Jackenmen) [inline redbot/cogs/mod/kickban.py]:
```suggestion
                    msg += _("\n\n**Reason:** {reason}").format(
```

File: redbot/cogs/mod/kickban.py [hunk 2]
```diff
@@ -638,16 +640,21 @@ async def tempban(
...
-            # We don't want blocked DMs preventing us from banning
-            msg = _("You have been temporarily banned from {server_name} until {date}.").format(
-                server_name=guild.name, date=unban_time.strftime("%m-%d-%Y %H:%M:%S")
-            )
-            if invite:
-                msg += _(" Here is an invite for when your ban expires: {invite_link}").format(
-                    invite_link=invite
-                )
-            await member.send(msg)
+        if settings["dm_on_kickban"]:
```
Comment (Jackenmen) [inline redbot/cogs/mod/kickban.py]: The message should be sent *always* but the reason should only be sent *if* dm_on_kickban is True

File: redbot/cogs/mod/kickban.py [hunk 3]
```diff
@@ -620,12 +620,14 @@ async def tempban(
             await ctx.send(_("I cannot do that due to Discord hierarchy rules."))
             return
 
+        settings = await self.config.guild(guild).all()
```
Comment (Jackenmen) [inline redbot/cogs/mod/kickban.py]: Can we go for `guild_data` as a var name here instead?

**Review by Jackenmen — CHANGES_REQUESTED** [review]

**Review by Jackenmen — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**